velox/functions/prestosql/ArrayFunctions.h

```diff
@@ -1125,6 +1125,28 @@ struct ArrayUnionFunction {
   }
 };
 
+/// This class implements the array_compact function.
+///
+/// DEFINITION:
+/// array_compact(x) -> array
+/// Remove all null elements from array x.
+template <typename T>
+struct ArrayCompactFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  // Results refer to the first input strings parameter buffer.
+  static constexpr int32_t reuse_strings_from_arg = 0;
+
+  template <typename Out, typename In>
+  void call(Out& out, const In& inputArray) {
+    for (const auto& item : inputArray) {
+      if (item.has_value()) {
```

Comment:
`mayHaveNulls` can help avoid one by one check if it returns false.

Reply:
ArrayCompactFunction already removed